From 99425c377c052c0ff450c1e903812081faa60471 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Thu, 27 Nov 2014 14:02:21 +0800
Subject: [PATCH 157/164] atomisp2: [WORKAROUND]Re-trigger capture in SDV mode
 when multiple buffers are queued

Due to the design of HALv3, sometimes in SDV mode HAL needs to
capture multiple images within one streaming cycle.
But the capture number cannot be determined by HAL.
So HAL only sets the capture number to be 1 and queue multiple
buffers. Atomisp driver needs to check this case and re-trigger
CSS to do capture when new buffer is queued.

Change-Id: I54738fa7387589160212edfce196188807cc3ebe
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-4029
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/30404
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_ioctl.c    |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 23b82d9..a7aa69b 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -1336,14 +1336,13 @@ done:
 	}
 
 	/* Workaround: Due to the design of HALv3,
-	 * sometimes in ZSL mode HAL needs to
+	 * sometimes in ZSL or SDV mode HAL needs to
 	 * capture multiple images within one streaming cycle.
 	 * But the capture number cannot be determined by HAL.
 	 * So HAL only sets the capture number to be 1 and queue multiple
 	 * buffers. Atomisp driver needs to check this case and re-trigger
 	 * CSS to do capture when new buffer is queued. */
 	if (asd->continuous_mode->val &&
-	    asd->run_mode->val == ATOMISP_RUN_MODE_PREVIEW &&
 	    atomisp_subdev_source_pad(vdev)
 	    == ATOMISP_SUBDEV_PAD_SOURCE_CAPTURE &&
 	    pipe->capq.streaming &&
-- 
1.7.9.5

